gitdb/db/loose.py

```diff
@@ -228,9 +229,22 @@ def store(self, istream):
                 rename(tmp_path, obj_path)
             # end rename only if needed
 
-            # make sure its readable for all ! It started out as rw-- tmp file
-            # but needs to be rwrr
-            chmod(obj_path, self.new_objects_mode)
+            # Ensure rename is actually done and file is stable
+            for _ in range(10):  # Retry up to 10 times
+                with suppress(PermissionError):
+                    # make sure its readable for all ! It started out as rw-- tmp file
+                    # but needs to be rwrr
+                    chmod(obj_path, self.new_objects_mode)
+                    break
+                time.sleep(0.1)
```

Comment:
Would this be better with an exponential backoff?

A set of 10 of these could be hard-coded (in milliseconds): `[1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289] .

Reply:
I quite like the idea. Done